client_test.go

```diff
@@ -157,3 +158,216 @@ func (suite *ClientTestSuite) TestUnauthorizedBoardFetch() {
 		assert.Equal(suite.T(), true, false)
 	}
 }
+
+// ==================================
+// ========== Boards.Create =========
+// ==================================
+
+// TestBadRequestBoardCreate tests that an error is appropriately thrown
+// when a board is attempted to be created without a name
+func (suite *ClientTestSuite) TestBadRequestBoardCreate() {
+	_, err := suite.client.Boards.Create("", &controllers.BoardCreateOptionals{})
+	assert.NotEqual(suite.T(), nil, err)
+
+	// Check error type
+	if pinterestError, ok := err.(*models.PinterestError); ok {
+		// Should be a 400
+		assert.Equal(suite.T(), http.StatusBadRequest, pinterestError.StatusCode)
+	} else {
+		// Make this error out, should always be a PinterestError
+		assert.Equal(suite.T(), true, false)
+	}
+}
+
+// TestTimeoutBoardCreate tests that an error is appropriately thrown
+// when a network timeout occurs
+func (suite *ClientTestSuite) TestTimeoutBoardCreate() {
+	_, err := suite.timeoutClient.Boards.Create("Some Board", &controllers.BoardCreateOptionals{})
+	assert.NotEqual(suite.T(), nil, err)
+}
+
+// TestUnauthorizedBoardCreate tests that an error is appropriately thrown
+// when the user makes an unauthorized request
+func (suite *ClientTestSuite) TestUnauthorizedBoardCreate() {
+	_, err := suite.unauthorizedClient.Boards.Create("Some Board", &controllers.BoardCreateOptionals{})
+	assert.NotEqual(suite.T(), nil, err)
+
+	// Check error type
+	if pinterestError, ok := err.(*models.PinterestError); ok {
+		// Should be a 401
+		assert.Equal(suite.T(), http.StatusUnauthorized, pinterestError.StatusCode)
+	} else {
+		// Make this error out, should always be a PinterestError
+		assert.Equal(suite.T(), true, false)
+	}
+}
+
+// ==================================
+// ========== Boards.Update =========
+// ==================================
+
+// TestTimeoutBoardUpdate tests that an error is appropriately thrown
+// when a network timeout occurs
+func (suite *ClientTestSuite) TestTimeoutBoardUpdate() {
+	_, err := suite.timeoutClient.Boards.Update("brandonrromano/go-pinterest-test", &controllers.BoardUpdateOptionals{})
+	assert.NotEqual(suite.T(), nil, err)
+}
+
+// TestUnauthorizedBoardUpdate tests that an error is appropriately thrown
+// when the user makes an unauthorized request
+func (suite *ClientTestSuite) TestUnauthorizedBoardUpdate() {
+	_, err := suite.unauthorizedClient.Boards.Update("brandonrromano/go-pinterest-test", &controllers.BoardUpdateOptionals{})
+	assert.NotEqual(suite.T(), nil, err)
+
+	// Check error type
+	if pinterestError, ok := err.(*models.PinterestError); ok {
+		// Should be a 401
+		assert.Equal(suite.T(), http.StatusUnauthorized, pinterestError.StatusCode)
+	} else {
+		// Make this error out, should always be a PinterestError
+		assert.Equal(suite.T(), true, false)
+	}
+}
+
+// TestForbiddenBoardUpdate tests that you may not update boards
+// that do not belong to you.
+func (suite *ClientTestSuite) TestForbiddenBoardUpdate() {
```

Comment:
Should rename this method to UnauthorizedBoardUpdate... As actually a 401 is returned